sleuth_backend/solr/models.py

```diff
@@ -0,0 +1,73 @@
+import json
+import time
+from . import connection as solr
+
+class SolrDocument(object):
+    """
+    An base class for documents that are inserted into Solr, an retured as 
+    search results from the Sleuth API.
+    """
+
+    def __init__(self, doc, **kwargs):
+        """
+        This method should be called by the subclass constructor.
+        """
+        self.doc = doc
+        for key in self.doc.keys():
+            if key in kwargs and key is not "type":
+                doc[key] = kwargs[key]
+
+    def output_format(self):
+        """
+        Returns the document in JSON format as it is supposed to be returned 
+        from a Sleuth API response. This method must be overridden.
+        """
+        pass
+
+    def save_to_solr(self, solr_connection):
+        """
+        Submits the document to the given Solr connection. This method should not
+        be overridden.
+        """
+        solr_connection.insert_document(self.type(), self.doc)
+
+    def type(self):
+        """
+        Returns a string representing the document type. This method should not
+        be overridden.
+        """
+        return self.doc["type"]
+
+class GenericPage(SolrDocument):
+    """
+    Represents a generic web page.
+    """
+    doc = {
+        "id": "",
+        "type": "genericPage",
+        "siteName": "",
+        "updatedAt": "",
+        "pageName": "",
+        "pageTitle": "",
+        "content": ""
```

Comment:
Are we not keeping the `keywords` field?

Reply:
Should be good for now, I'll add it in later if I can get something good working